src/hooks/useJoinedColoniesWithExtensions.ts

```diff
@@ -0,0 +1,26 @@
+import { ADDRESS_ZERO } from '~constants/index.ts';
+import { useGetAllColoniesExtensionsQuery } from '~gql';
+import { notNull } from '~utils/arrays/index.ts';
+
+const sortByDate = <T extends { createdAt: string }>(a: T, b: T) =>
+  new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime();
+
+const useJoinedColoniesWithExtensions = (userAddress?: string) => {
```

Comment:
This hook seems very specific to `useNetworkMotionStates`, I would either make it a helper or place it next to that hook to indicate it's not intended to be reusable